src/NUnit.TestAdapter.Tests.Acceptance/Utils.cs

```diff
@@ -0,0 +1,215 @@
+                    case '\\':
+                    case '/':
+                        r.Append(allowDirectorySeparators ? Path.DirectorySeparatorChar : '-');
+                        break;
+                    case '\0':
+                    case '\f':
+                    case '?':
+                        break;
+                    case '\t':
+                    case '\n':
+                    case '\r':
+                    case '\v':
+                        r.Append(' ');
+                        break;
+                    default:
+                        r.Append('_');
+                        break;
+                }
+
+                i = replaceIndex + 1;
+                replaceIndex = arbitraryString.IndexOfAny(invalidChars, i);
+            } while (replaceIndex != -1);
+
+            r.Append(arbitraryString, i, arbitraryString.Length - i);
+
+            return r.ToString();
+        }
+
+        public static string CreateUniqueDirectory(string parentDirectory, string name = null)
+        {
+            parentDirectory = Path.GetFullPath(parentDirectory);
+
+            var safeName = name is null ? null : GetSafeFilename(name);
+
+            var path = Path.Combine(parentDirectory, safeName ?? "1");
+
+            if (Directory.Exists(path))
+            {
+                for (var id = 2; ; id++)
+                {
+                    path = Path.Combine(parentDirectory, safeName is null ? id.ToString() : safeName + "_" + id);
+                    if (!Directory.Exists(path)) break;
+                }
+            }
+
+            Directory.CreateDirectory(path);
+            return path;
+        }
+
+        public static void DeleteDirectoryRobust(string directory)
+        {
+            for (var attempt = 1; ; attempt++)
+            {
+                try
+                {
+                    Directory.Delete(directory, recursive: true);
+                    break;
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    break;
+                }
+                catch (IOException ex) when (attempt < 3 && (WinErrorCode)ex.HResult == WinErrorCode.DirNotEmpty)
+                {
+                    TestContext.WriteLine("Another process added files to the directory while its contents were being deleted. Retrying...");
+                }
+            }
+        }
+
+        private enum WinErrorCode : ushort
+        {
+            DirNotEmpty = 145
+        }
+
+        /// <summary>
+        /// Removes any indentation that is common to all lines. If the first line has no indentation,
+        /// indentation common to all other lines is removed. If the first line is empty, it is removed.
+        /// Empty lines are not considered when calculating indentation.
+        /// </summary>
+        public static string RemoveIndent(string indented)
+        {
+            if (indented is null) return null;
+
+            var reader = new StringReader(indented);
```

Comment:
Should this be disposed?

EDIT: I should have read https://docs.microsoft.com/en-us/dotnet/api/system.io.stringreader?view=netframework-4.7.2#remarks 😄

Reply:
I never know. In theory they could start using a buffer pool internally and need you to dispose to return the rented buffers, but in practice there is no way they would ever do that after 18 years of telling people that the type doesn't need to be disposed. The only benefit would be consistency, in case we change it to a type that does need disposal. I'm only using it as a newline-seeking aid, so it seems more likely if anything that it would become a custom struct if we changed the type.